M0090IDE-3982 document resolve in workspace (#443)
MicroEJ/docs: SDKUserGuide/resolveDependenciesInWorkspace.rst

the project is directly used for compilation and execution instead of using the dependency,
provided that the dependency's project has the same version as the one required by the module.

For example, suppose that the workspace contains a module ``myApp`` and its dependency ``mylib`` :

.. figure:: images/dependencyInWorkspace.png
   :alt:  A module and its dependency opened in the same workspace
   :align: center
   :scale: 100%

   A module and its dependency opened in the same workspace

If the ``mylib`` project's version is ``1.0.0``, it is used for compilation and execution.
Otherwise the published artifact is downloaded from the artifact repository.

To avoid a dependency to be resolved in the workspace, you can close the corresponding project or remove it from the workspace.

.. warning::
   If you open, close, import or remove a project, you must refresh the dependency resolution of other previously imported projects
   by clicking on the :guilabel:`Resolve All` button :

   .. figure:: images/mmm_shortcut_resolveAll.png
      :alt: Resolve all the workspace projects
      :align: center
      :scale: 100%
   
      Resolve all the workspace projects

> .. _resolve_foundation_library_in_workspace:

Reviewer: shall we pull-up the sections at the same level of Dependencies ?

![image](https://user-images.githubusercontent.com/19801840/191685445-34024e80-5e51-4368-a79e-d25d6619cb0b.png)

shall we pull up

Author: Done.

> Resolve Foundation Library in workspace

Reviewer: ```suggestion
Resolve Foundation Library in Workspace
```

> Resolve Foundation Library in workspace

Reviewer: Sync with the previous title: either move to the plural form (Foundation Libraries) or move Dependencies to singular form (Dependency) ?

Author: Done, I chose to keep the plural form.

---------------------------------------

A :ref:`Foundation Library <libraries>` is composed of : 

> - A MicroEJ API project that contains API skeletons used at compile time
>   
> - A MicroEJ Implementation project that contains the runtime code executed by the Platform and Low Level C header files
>   
> - The C code that contains the implementation of native methods linked to the C project
>   
> - A Mock project that contains the implementation of native methods for simulation

Reviewer: The How-To is confusing: a Foundation Library does not include the C implementation code and the Mock.

```suggestion
- An API project that contains Java classes, methods and fields used at compile time with their associated Javadoc,
  
- An Implementation project that contains the runtime code executed by the Platform and Low Level C header files

Beside Foundation Library projects, there is usually a :ref:`Mock <mock>` project that contains the implementation of native methods for simulation.
```

(only focus on the mock project since this is the purpose of this section)

.. note::

>    To learn how to setup a Foundation Library, please consult our MicroEJ How-to available on 

Reviewer: ```suggestion
   To learn how to setup a Foundation Library, please consult the How-to available on 
```

   `<https://github.com/MicroEJ/How-To/tree/master/FoundationLibrary-Get-Started>`_.

> When the API is set as a dependency, the Implementation project and the Mock project are automatically used for execution on Simulator 

Reviewer: ```suggestion
When the API is set as a dependency, the Implementation project and the Mock project are automatically used for execution on Simulator 
```

We must separate:

- the Implementation project: it is also taken into account when building the Application on Device 
- the mock project; only on Simulation

Author: Done.

> When the API is set as a dependency, the Implementation project and the Mock project are automatically used for execution on Simulator 

Reviewer: Also add:
- in-place prior to the Platform implementation

Add Warning: the the implementation project must have been built in the Platform once.
What about the mock project ?

Author: I have done some tests and I didn't have to build the Implementation project and the Mock project in the Platform, they were automatically used, so I didn't add Warning.

Reviewer: OK my mistake !
